airflow/dag_processing/manager.py

```diff
@@ -1042,8 +1040,7 @@ def _collect_results_from_processor(self, processor) -> None:
             run_count=self.get_run_count(processor.file_path) + 1,
         )
         self._file_stats[processor.file_path] = stat
-
-        file_name = os.path.splitext(os.path.basename(processor.file_path))[0].replace(os.sep, ".")
```

Comment:
The final `replace` call is so weird… Since `basename` is called, the value just can never contain an `os.sep` in the first place.